bucketlist/app.py

```diff
@@ -0,0 +1,281 @@
+                bucketlist_name=bucketname).first()
+            if bucket is not None:
+                return({'message': 'Bucketlist already exists'}, 400)
+            bucketlist = Bucketlist(bucketlist_name=bucketname,
+                                    date_created=datetime.datetime.now(),
+                                    date_modified=datetime.datetime.now(),
+                                    created_by=g.user.id)
+            bucketlist.add(bucketlist)
+            return({'message': 'Bucketlist successfully created'}, 201)
+        else:
+            return({'message': 'Bucketlist name cannot be blank'}, 400)
+
+
+class BucketList(Resource):
+    '''Retrievs specific bucketlist using an id'''
+    @auth.login_required
+    def get(self, id):
+        bucketlists = Bucketlist.query.filter_by(
+            id=id, created_by=g.user.id).first()
+        if bucketlists:
+            return marshal(bucketlists, Bucketlist_marshaller)
+        else:
+            return ({'message': 'Bucketlist does not exist'}, 404)
+
+    @auth.login_required
+    def put(self, id):
+        '''Updates a specific bucketlist using the id'''
+        # Queries the db to ensure that search a bucketlist exists
+        bucketlist = Bucketlist.query.filter_by(
+            id=id, created_by=g.user.id).first()
+        if bucketlist:
+            parser = reqparse.RequestParser()
+            parser.add_argument('Bucketname', required=True,
+                                help="Username cannot be blank")
+            update = parser.parse_args()
+            bname = update['Bucketname']
+            bucketlist.bucketlist_name = bname
+            bucketlist.update()
+            return({'message': 'Bucketlist Updated'}, 201)
+        else:
+            return({'message': 'Bucketlist does not exist'}, 404)
+
+    @auth.login_required
+    def delete(self, id):
+        '''deletes specific bucketlist as per the id provided'''
+        bucketlist = Bucketlist.query.filter_by(
+            id=id, created_by=g.user.id).first()
+        if bucketlist:
+            bucketlist.delete(bucketlist)
+            return ({'message': 'Bucketlist deleted'})
+        else:
+            return ({'message': 'Bucketlist does not exist'}, 404)
+
+
+class BucketListItems(Resource):
+    @auth.login_required
+    def post(self, id):
+        '''Adds a item to a bucketlist id given'''
+        parser = reqparse.RequestParser()
+        parser.add_argument('item_name', required=True,
+                            help='Item name cannot be blank')
+
+        item_details = parser.parse_args()
+        item_name = item_details['item_name']
+        if len(item_name) > 0:
+            item = BucketListItem(item_name=item_name,
+                                  bucketlist_name=id,
+                                  date_created=datetime.datetime.now(),
+                                  date_modified=datetime.datetime.now(),
+                                  done=False)
```

Comment:
`date_created`, `date_modified` should be automatically populated by the model. You can set done as `False` by default so that you don't have to add when creating a bucketlist item.